network.c

```diff
@@ -1323,6 +1323,69 @@ static ssize_t network_read_line(struct iio_context_pdata *pdata,
 {
 	bool found = false;
 	size_t i;
+#ifdef __linux__
+	struct iio_network_io_context *io_ctx = io_data;
+	ssize_t ret;
+	size_t bytes_read = 0;
+
+	do {
+		size_t to_trunc;
+
+		ret = network_recv(io_ctx, dst, len, MSG_PEEK);
+		if (ret < 0)
+			return ret;
+
+		/* Lookup for the trailing \n */
+		for (i = 0; i < (size_t) ret && dst[i] != '\n'; i++);
+		found = i < (size_t) ret;
+
+		len -= ret;
+		dst += ret;
+
+		if (found)
+			to_trunc = i + 1;
+		else
+			to_trunc = (size_t) ret;
+
+		/* Advance the read offset to the byte following the \n if
+		 * found, or after the last charater read otherwise */
+		ret = network_recv(io_ctx, NULL, to_trunc, MSG_TRUNC);
+		if (ret < 0)
+			return ret;
+
+		bytes_read += to_trunc;
+	} while (!found && len);
+
+	if (!found)
+		return -EIO;
+	else
+		return bytes_read;
+#else
+	for (i = 0; i < len - 1; i++) {
+		ssize_t ret = network_read_data(pdata, io_data, dst + i, 1);
+
+		if (ret < 0)
+			return ret;
+
+		if (dst[i] != '\n')
+			found = true;
+		else if (found)
+			break;
+	}
+
+	if (!found || i == len - 1)
+		return -EIO;
+
+	return (ssize_t) i + 1;
+#endif
+}
+
+/* use for systems that don't support MSG_TRUNC (e.g. Windows Subsystem for Linux) */
+static ssize_t network_read_line_no_trunc (struct iio_context_pdata *pdata,
+		void *io_data, char *dst, size_t len)
```

Comment:
This is a bit too much copy&paste for my taste. If we update this code in the future we have to update the same code in two places. Maybe you can factor this out into a common function that takes a extra do_trunc boolean flag that is set to either true or false. And then let the compiler take care of optimizing this as it seems fit.